src/python/pants/backend/shell/target_types.py

```diff
@@ -192,6 +195,77 @@ async def generate_targets_from_shell_library(
     )
 
 
+# -----------------------------------------------------------------------------------------------
+# `shell_command` target
+# -----------------------------------------------------------------------------------------------
+
+
+class ShellCommandCommandField(StringField):
+    alias = "command"
+    required = True
+    help = (
+        "Shell command to execute.\n\n" "The command is executed as 'bash -c <command>' by default."
+    )
+
+
+class ShellCommandOutputsField(StringSequenceField):
+    alias = "outputs"
+    help = (
+        "Specify the shell command output files and directories.\n\n"
+        "Use a trailing slash on directory names, i.e. `my_dir/`."
+    )
+
+
+class ShellCommandSources(ShellSources):
+    # We solely register this field for codegen to work.
+    alias = "_sources"
+
+
+class ShellCommandToolsField(StringSequenceField):
+    alias = "tools"
+    required = True
+    help = (
+        "Specify required executable tools that might be used.\n\n"
+        "Only the tools explicitly provided will be available on the search PATH."
+    )
+
+
+class ShellCommandLogOutputField(BoolField):
+    alias = "log_output"
+    default = False
+    help = "Set to true if you want the output from the command logged to the console."
+
+
+class ShellCommand(Target):
+    alias = "experimental_shell_command"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        Dependencies,
+        ShellCommandCommandField,
+        ShellCommandLogOutputField,
+        ShellCommandOutputsField,
+        ShellCommandSources,
+        ShellCommandToolsField,
+    )
+    help = dedent(
+        """\
+        Execute any external tool for its side effects.
+        This may be retried and/or cancelled, so ensure that it is idempotent.
```

Comment:
Nit, should probably move below the example. This is a nuanced detail that doesn't explain _what_ this target is, only a limitation when using it. The example is really helpful to figure out _what_ it is.

Might also be worth adding a paragraph below the example explaining how you can depend on `shell_library` targets to run its script, or you can directly invoke Bash commands like `touch my_file.ext`.

Reply:
Oh! Another key detail missing from this help, you must add this target to the `dependencies` of each consumer, such as your `python_tests`. When relevant, Pants will run your `command` and insert the `output_files` into that build's context.

Reply:
> Might also be worth adding a paragraph below the example explaining how you can depend on shell_library targets to run its script

Oh, wait, I don't think I fully understand what you mean here.

Reply:
Oh, could we get around this limitation, at least to some extent, if we implement the `GenerateTargetsRequest` union for shell_command as well (and generate file targets for any output files)?

Reply:
IMO, only being able to bring in all of the output files of the script is totally fine for now. And `files` is almost certainly the right output type.

`GenerateTargetsRequest` wouldn't be able to fully determine which files existed without actually running the script, because `directory/` entries include the children of the directory.

Reply:
> Oh, wait, I don't think I fully understand what you mean here.

What I mean is that if you set your command to be `./my_script.sh`, you should add a `shell_library` target which includes `my_script.sh` in its `sources` and include that in the `dependencies` field.